src/interfaces/records-read.ts

```diff
@@ -57,6 +67,32 @@ export class RecordsRead extends Message<RecordsReadMessage> {
     return new RecordsRead(message);
   }
 
+  /**
+   * Creates a Records interface Filter using either the `recordId` or `protocol` & `protocolPath` of the incoming Message.
+   * Also filters on an optional `parentId` for `protocolPath` based reads
+   * @param descriptor message descriptor with optional properties `recordId`, `protocol` and `protocolPath`
+   *
+   * @returns {Filter} with a Records interface as well as the appropriate filter params
+   * @throws {DwnError} if `recordId` or `protocol` and `protocolPath` are missing
```

Comment:
```suggestion
   * @throws {DwnError} if `recordId` and either `protocol` or `protocolPath` are missing
```

Reply:
Yeah this was worded poorly by me, ended up going with:
```
@throws {DwnError} when either `recordId` is missing, or both `protocol` and `protocolPath` are missing.
```

I think that's a better description.